src/test/java/io/cdap/plugin/ariba/source/AribaRecordReaderTest.java

```diff
@@ -0,0 +1,337 @@
+public class AribaRecordReaderTest {
+
+  private AribaPluginConfig pluginConfig;
+  private AribaServices aribaServices;
+  private MockPipelineConfigurer pipelineConfigurer;
+  private static ObjectMapper mapper = new ObjectMapper();
+
+  @Before
+  public void setUp() {
+    pipelineConfigurer = new MockPipelineConfigurer(null);
+    pluginConfig = new AribaPluginConfig("unit-test-ref-name", "https://openapi.au.cloud.ariba.com",
+                                         "prod", "CloudsufiDSAPP-T",
+                                         "SourcingProjectFactSystemView", "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                         "c3B5wvrEsjKucFGlGhKSWUDqDRGE2Wds", "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+  }
+
+  @Test
+  public void testInitialize() throws IOException, AribaException, InterruptedException {
+    aribaServices = new AribaServices(pluginConfig);
+    AribaRecordReader aribaRecordReader = new AribaRecordReader(aribaServices, pipelineConfigurer.getOutputSchema());
+    AribaInputSplit aribaInputSplit = new AribaInputSplit("sourceView.zip", "3343ddsfsg3434");
+    JsonNode nodeRecord = mapper.readTree(rowData);
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.getAccessToken();
+        result = "testToken";
+        minTimes = 0;
+
+        aribaServices.fetchData(anyString, anyString);
+        result = nodeRecord;
+        minTimes = 0;
+      }
+    };
+    aribaRecordReader.initialize(aribaInputSplit, null);
+    aribaRecordReader.getProgress();
+    aribaRecordReader.close();
+    Assert.assertEquals(aribaRecordReader.getCurrentKey(), NullWritable.get());
+    Assert.assertTrue(aribaRecordReader.nextKeyValue());
+  }
+
+  @Test
+  public void testNextKeyValueFalse(@Mocked AribaInputSplit aribaInputSplit,
+                                    @Mocked JsonNode node) throws IOException, AribaException, InterruptedException {
+    aribaServices = new AribaServices(pluginConfig);
+    AribaRecordReader aribaRecordReader = new AribaRecordReader(aribaServices, pipelineConfigurer.getOutputSchema());
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.getAccessToken();
+        result = "testToken";
+        minTimes = 0;
+
+        aribaServices.fetchData(anyString, anyString);
+        result = node;
+        minTimes = 0;
+      }
+    };
+    aribaRecordReader.initialize(aribaInputSplit, null);
+    Assert.assertFalse(aribaRecordReader.nextKeyValue());
+  }
+
+  @Test
+  public void testReadFields() throws IOException {
+    AribaStructuredTransformer aribaStructuredTransformer = new AribaStructuredTransformer();
+    JsonNode nodeRecord = mapper.readTree(rowData);
+    StructuredRecord structuredRecord = aribaStructuredTransformer.readFields(nodeRecord, getPluginSchema());
+    Assert.assertEquals("AribaColumnMetadata", structuredRecord.getSchema().getRecordName());
```

Comment:
can we validate the contents of the schema?

Reply:
Fixed